### PR TITLE
Add the missing node for idam-app service

### PR DESCRIPTION
**Pull request by jasonpaige**
The idam-app microservice has been missing and so relationships between
services has been miss-represented.

I could do with a hand double checking my assumptions on what talks to what with regards to idam please.

### CODE REVIEW

File: docs/microservices.json [hunk 1]
```diff
@@ -68,12 +73,17 @@
                     "apis": []
                 },
                 {
-                    "id": "idam-web",
+                    "id": "authentication-web",
+                    "hard": true,
+                    "apis": []
+                },
```
Comment (rkondratowicz) [inline docs/microservices.json]: does CCD API Gateway talk to idam frontend?

File: docs/microservices.json [hunk 2]
```diff
@@ -320,7 +393,7 @@
             "spec": null,
             "dependencies": [
                 {
-                    "id": "idam-web",
+                    "id": "authentication-web",
```
Comment (rkondratowicz) [inline docs/microservices.json]: probably idam-app as well... not sure though

File: docs/microservices.json [hunk 3]
```diff
@@ -410,7 +483,7 @@
                     "apis": []
                 },
                 {
-                    "id": "idam-web",
+                    "id": "authentication-web",
```
Comment (rkondratowicz) [inline docs/microservices.json]: also idam-app (for token validation)

File: docs/microservices.json [hunk 4]
```diff
@@ -452,7 +525,7 @@
...
```
Comment (rkondratowicz) [inline docs/microservices.json]: also idam-app

**Comment by timja-kainos** [comment]
Also missing cmc-claim-store and legal-rep frontend talking to document management

WIP integration to ccd as well..

**Comment by jasonpaige** [comment]
can u add those CMC working parts in another PR @timja-kainos ? this thing will take off if we have someone from each team looking after their own domains. I'm half guessing at a lot of stuff and I'm too far removed to be that accurate (as we can see from this PR :P)

**Comment by timja-kainos** [comment]
Can do @jasonpaige

**Comment by rkondratowicz** [comment]
bonus points for publishing swagger spec ;)

File: docs/microservices.json (hunk 1 above)
Comment (jasonpaige) [reply docs/microservices.json]: I don't think so. Their front end is an angular app which talks to the API gateway

Comment (rkondratowicz) [reply docs/microservices.json]: so this should be removed then

Comment (jasonpaige) [reply docs/microservices.json]: I'd need @vlaurin to confirm but iirc this gateway does some IdAM user verification for them?

File: docs/microservices.json (hunk 2 above)
Comment (jasonpaige) [reply docs/microservices.json]: yup, I think you're right

File: docs/microservices.json (hunk 1 above)
Comment (rkondratowicz) [reply docs/microservices.json]: yeah, maybe `idam-app`. definitely not the web frontend.

File: docs/microservices.json (hunk 3 above)
Comment (jasonpaige) [reply docs/microservices.json]: ja

File: docs/microservices.json (hunk 4 above)
Comment (jasonpaige) [reply docs/microservices.json]: 👍

File: docs/microservices.json (hunk 1 above)
Comment (vlaurin) [reply docs/microservices.json]: Yes, CCD's gateway talks to `idam-app` and `service-auth-provider` but not to the IDAM front-ends

**Comment by rkondratowicz** [comment]
should this PR be updated or just removed?

**Pull request closed without merging**